packages/compass-schema/src/stores/store.js

```diff
@@ -217,6 +222,62 @@ const configureStore = (options = {}) => {
       }
     },
 
+    _calculateDepthByPath(input, paths) {
+      if (!input) {
+        return paths;
+      }
+      input.forEach(({types, fields, path, bsonType}) => {
+        const increment = bsonType === 'Array' ? 1 : 0;
+        const score = path.split('.').length + increment;
+        if (!paths[path] || paths[path] < score) {
+          paths[path] = score;
+        }
+        this._calculateDepthByPath(types ?? fields, paths);
+      });
+
+      return paths;
+    },
+
+    calculateSchemaDepth(schema) {
+      const response = this._calculateDepthByPath(schema.fields, {});
+      const values = Object.values(response);
+      return Math.max(...values, ...[0]);
```

Comment:
You probably meant

```suggestion
      return Math.max(...values, 0);
```

or

```suggestion
      return Math.max(...[...values, 0]);
```

here, right? I mean, it’s the same, but `...[0]` looks a bit odd :slightly_smiling_face:

Reply:
absolutely!